tests/system/test_alembic.py

```diff
@@ -146,23 +158,39 @@ def test_alembic_scenario(alembic_table):
         bigquery_time_partitioning=TimePartitioning(field="transaction_time"),
     )
 
+    _prevent_rate_limiting()
     op.alter_column("transactions", "amount", nullable=True)
     assert alembic_table("transactions", "schema") == [
         SchemaField("account", "INTEGER", "REQUIRED"),
         SchemaField("transaction_time", "DATETIME", "REQUIRED"),
         SchemaField("amount", "NUMERIC(11, 2)"),
     ]
 
+    _prevent_rate_limiting()
     op.create_table_comment("transactions", "Transaction log")
     assert alembic_table("transactions").description == "Transaction log"
 
+    _prevent_rate_limiting()
     op.drop_table("transactions")
 
     # Another thing we can do is alter the datatype of a nullable column,
     # if allowed by BigQuery's type coercion rules
+    _prevent_rate_limiting()
     op.create_table("identifiers", Column("id", Integer))
 
+    _prevent_rate_limiting()
     op.alter_column("identifiers", "id", type_=Numeric)
     assert alembic_table("identifiers", "schema") == [SchemaField("id", "NUMERIC")]
 
+    _prevent_rate_limiting()
     op.drop_table("identifiers")
+
+
+def _prevent_rate_limiting():
+    """
+    Adds a delay between operations to prevent exceeding the "Maximum rate of table metadata update operations per table".
+
+    The rate limit is set to a maximum of 5 operations per 10 seconds. By introducing a sleep of 2 seconds
```

Comment:
I'm somewhat surprised that our exponential backoff query job retry logic doesn't handle this already. Do we have a link to a failing test stacktrace?

Instead of sleep, I'd like for us to add the rate limit as an allowed job retry reason in

https://github.com/googleapis/python-bigquery/blob/d1161dddde41a7d35b30033ccbf6984a5de640bd/google/cloud/bigquery/retry.py#L76-L84

if we know that the error is safe to retry.

If we do go this route, let's add a link to https://cloud.google.com/bigquery/quotas#standard_tables where this limit is defined.

Reply:
@tswast 

Sure, it would be best handled in the `python-bigquery` code.. The work around I suggested is a quite quick and dirty solution..

Here's a stacktrace of the error https://source.cloud.google.com/results/invocations/ffafb866-6bc0-423f-a86b-df69fb270d57/log.

Reply:
I've filed https://github.com/googleapis/python-bigquery/issues/1790 as the preferred approach to fixing this problem.

Reply:
I am inclined to see if we can get this fixed.
Will work through things on the Issue Tim created...https://github.com/googleapis/python-bigquery/issues/1790